### ISSUE TITLE
[Question] Training  with default config for fconv stopping after epoch 12 

### ISSUE THREAD

**Issue by panwarnaveen9** [facebookresearch/fairseq-lua#73]
I am training the model with default command given at git. 
`fairseq train -sourcelang de -targetlang en -datadir data-bin/iwslt14.tokenized.de-en -model fconv -nenclayer 4 -nlayer 3 -dropout 0.2 -optim nag -lr 0.25 -clip 0.1  -momentum 0.99 -timeavg -bptt 0 -savedir trainings/fconv`
Training is stopping at epoch 12 and it's calculating bleu score after checkpoint 13. 
But default `maxepoch` is 100

**Comment by jgehring** [comment]
Hi @panwarnaveen9, this is due to early stopping. For the nag optimizer, when validation perplexity stops improving the learning rate is divided by `-lrshrink` at every epoch (you should also see this in the log) until it falls below `minlr`, which is hard-coded to `1e-5` for nag. You can change `config.minlr` in [train.lua:263](https://github.com/facebookresearch/fairseq/blob/master/train.lua#L263) if you think that training should go on for a longer period of time.
